### PR TITLE
avocado.utils.pci: Couple of adjustments to make params explicite

### PR DESCRIPTION
**Pull request by ldoktor**
The first 2 commits are bugfixes and part of a different PR. The 3rd commit is a RFC regarding function name and the last commit should clarify what kind of parameter is expected in each function.

### CODE REVIEW

**Comment by narasimhan-v** [comment]
@ldoktor Thanks a lot for these documentation improvements and fixes :)

File: avocado/utils/pci.py [hunk 1]
```diff
@@ -59,21 +59,23 @@ def get_pci_addresses():
         return addresses
 
 
-def get_num_devices_in_domain(domain):
+def get_num_interfaces_in_pci(dom_pci_address):
```
Comment (ldoktor) [inline avocado/utils/pci.py]: Well there are human-interface-devices, there are infrared-interfaces, there are ata interfaces and one of the interfaces is network interface, so I don't associate interface necessarily with network.

On the other hand for example my integrated graphic card `00:02.0`, which is a single device, provides numerous interfaces like framebuffer, i2c, backlight, control, ....

I'm not sure whether `interface` is the best word, but I believe `device` is definitely incorrect. That's why I sent those changes in separate to get some responses on this...

**Comment by ldoktor** [comment]
No problem, it comes from the review and it was not crucial to the function, so I rather did it after merging it. Btw I'll send a v2 without the first 2 commits when we get a consensus on the function name.

**Comment by ldoktor** [comment]
@narasimhan-v I looked at http://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/tree/Documentation/ABI and most docs starts with something like "this interface" or "this is an interface between..." so I think `interface` is actually fitting. Anyway let me know what do you think about this, I don't think it's necessary to rush things, but I'd like to have a clear, well defined name before the next release (which is ~2-4 weeks due to Christmas).

File: avocado/utils/pci.py (hunk 1 above)
Comment (narasimhan-v) [inline avocado/utils/pci.py]: Sure, `interface` makes sense then.

File: avocado/utils/pci.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ def get_num_devices_in_domain(domain):
     :return: number of devices in a pci domain.
     """
     cmd = "ls -l /sys/class/*/ -1"
-    output = process.system_output(cmd, ignore_status=True)
+    output = process.system_output(cmd, ignore_status=True, shell=True)
```
Comment (apahim) [inline avocado/utils/pci.py]: Ok

File: avocado/utils/pci.py [hunk 3]
```diff
@@ -133,7 +133,7 @@ def get_slot_from_sysfs(pci_address):
 
     :return: slot of pci address from sysfs.
     """
-    if not os.path.isdir('/sys/bus/pci/devices/%s' % pci_address):
+    if not os.path.isdir('/sys/bus/pci/devices/%s/devspec' % pci_address):
```
Comment (apahim) [inline avocado/utils/pci.py]: ok

File: avocado/utils/pci.py [hunk 4]
```diff
@@ -59,21 +59,23 @@ def get_pci_addresses():
...
     """
     cmd = "ls -l /sys/class/*/ -1"
     output = process.system_output(cmd, ignore_status=True, shell=True)
     if output:
-        domain = '/%s' % domain
+        filt = '/%s' % dom_pci_address
         count = 0
         for line in output.splitlines():
-            if domain in line:
+            if filt in line:
```
Comment (apahim) [inline avocado/utils/pci.py]: ok

File: avocado/utils/pci.py [hunk 5]
```diff
@@ -292,28 +293,30 @@ def get_vpd(pci_addr):
...
     :return: dictionary of cfg data of a pci address.
     """
-    cmd = "lscfg -vl %s" % pci_addr
+    cmd = "lscfg -vl %s" % dom_pci_address
     cfg = process.system_output(cmd)
     cfg_dic = {}
-    desc = re.match(r'  (%s)( [-\w+,\.]+)+([ \n])+([-\w+, \(\)])+' % pci_addr,
-                    cfg).group()
+    desc = re.match(r'  (%s)( [-\w+,\.]+)+([ \n])+([-\w+, \(\)])+'
+                    % dom_pci_address, cfg).group()
```
Comment (apahim) [inline avocado/utils/pci.py]: ok

**Comment by apahim** [comment]
@ldoktor could you please rebase? I will merge right away then,

**Pull request closed without merging**